src/test/java/co/cask/cdc/plugins/performance/CDCPipelinePT.java

```diff
@@ -0,0 +1,245 @@
+
+    Map<String, String> sinkProps = ImmutableMap.<String, String>builder()
+      .put(CDCBigTableConfig.PROJECT, BIGTABLE_PROJECT)
+      .put(CDCBigTableConfig.INSTANCE, BIGTABLE_INSTANCE)
+      .put(CDCBigTableConfig.SERVICE_ACCOUNT_FILE_PATH, BIGTABLE_SERVICE_ACCOUNT_FILE_PATH)
+      .put(Constants.Reference.REFERENCE_NAME, "CDCBigTableSink")
+      .build();
+    ETLPlugin sinkConfig = new ETLPlugin("CDCBigTable", SparkSink.PLUGIN_TYPE, sinkProps);
+
+    programManager = deployETL(sourceConfig, sinkConfig, APP_NAME);
+  }
+
+  @After
+  @Override
+  public void tearDown() throws Exception {
+    if (programManager != null) {
+      programManager.stop();
+      programManager.waitForStopped(10, TimeUnit.SECONDS);
+      programManager.waitForRun(ProgramRunStatus.KILLED, 10, TimeUnit.SECONDS);
+    }
+    super.tearDown();
+    if (connection != null) {
+      connection.close();
+    }
+  }
+
+  @Test
+  public void testSqlServerToBigtablePipeline() throws Exception {
+    long testStart = System.currentTimeMillis();
+    LOG.info("Starting pipeline");
+    programManager.startAndWaitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    LOG.info("Waiting until {} records are present in Bigtable", TEST_DATA_INSERTS);
+
+    Awaitility.await()
+      .atMost(TEST_TARGET_TABLE_CREATED_TIMEOUT_SECONDS, TimeUnit.SECONDS)
+      .pollInterval(Duration.TEN_SECONDS)
+      .ignoreException(StatusRuntimeException.class)
+      .untilAsserted(() -> {
+        TableName tableName = TableName.valueOf(dbTableName);
+        Assert.assertTrue(String.format("Table '%s' was not created", tableName),
+                          connection.getAdmin().tableExists(tableName));
+        Assert.assertTrue(connection.getAdmin().isTableAvailable(tableName));
+        Assert.assertTrue(connection.getAdmin().isTableEnabled(tableName));
+    });
+
+    Awaitility.await()
+      .atMost(TEST_DATA_TRANSFERED_TIMEOUT_SECONDS, TimeUnit.SECONDS)
+      .pollInterval(Duration.TEN_SECONDS)
+      .ignoreException(IOExceptionWithStatus.class)
+      .untilAsserted(() -> {
+        TableName tableName = TableName.valueOf(dbTableName);
+        Table table = connection.getTable(tableName);
+        int rowCount = getRowCount(table);
+        LOG.info("Currently {} records are present in Bigtable", rowCount);
+        Assert.assertEquals(TEST_DATA_INSERTS, rowCount);
+      });
+
+    long testEnd = System.currentTimeMillis();
+    long elapsedSeconds = (testEnd - testStart) / 1000;
+    long recordsPerSecond = TEST_DATA_INSERTS / elapsedSeconds;
+    LOG.info("Test finished. Transferred '{}' records. Elapsed time is '{} seconds' ({} records/second)",
+             TEST_DATA_INSERTS, elapsedSeconds, recordsPerSecond);
+  }
+
+  private static int getRowCount(Table table) throws IOException {
+    int rowCount = 0;
+    try (ResultScanner scanner = table.getScanner(new Scan())) {
+      for (Result rs = scanner.next(); rs != null; rs = scanner.next()) {
```

Comment:
is there a different function besides scan which can be used to get the row count and is more efficient? If not, prefer while(rs.hasNext()) {
  rs.next();
  rowCount++;
}

Reply:
there is an AggegationClient for HBase but it is not supported for Bigtable. Also there is no hasNext method for ResultScanner.

Reply:
sg